engine/access/ingestion/engine_test.go

```diff
@@ -334,44 +319,48 @@ func (s *Suite) TestOnFinalizedBlockSeveralBlocksAhead() {
 		s.finalizedBlock = block.Header
 	}
 
+	// latest of all the new blocks which are newer than the last block processed
+	latestBlock := blocks[2]
+
 	// block several blocks newer than the last block processed
 	hotstuffBlock := hotmodel.Block{
-		BlockID: blocks[2].ID(),
-	}
-	// for each of the guarantees, we should request the corresponding collection once
-	needed := make(map[flow.Identifier]struct{})
-	for _, guarantee := range blocks[0].Payload.Guarantees {
-		needed[guarantee.ID()] = struct{}{}
+		BlockID: latestBlock.ID(),
 	}
 
+	requestCountPerBlock := 4
 	wg := sync.WaitGroup{}
-	wg.Add(4)
+	wg.Add(requestCountPerBlock * newBlocksCount)
 
-	s.request.On("EntityByID", mock.Anything, mock.Anything).Run(
-		func(args mock.Arguments) {
-			collID := args.Get(0).(flow.Identifier)
-			_, pending := needed[collID]
-			s.Assert().True(pending, "collection should be pending (%x)", collID)
-			delete(needed, collID)
-			wg.Done()
-		},
-	)
+	// expected all new blocks after last block processed
+	for _, block := range blocks {
+		s.blocks.On("IndexBlockForCollections", block.ID(), []flow.Identifier(flow.GetIDs(block.Payload.Guarantees))).Return(nil).Once()
 
-	// expected next block after last block processed
-	s.blocks.On("IndexBlockForCollections", blocks[0].ID(), []flow.Identifier(flow.GetIDs(blocks[0].Payload.Guarantees))).Return(nil).Once()
-	s.results.On("Index", mock.Anything, mock.Anything).Return(nil)
+		for _, cg := range block.Payload.Guarantees {
+			s.request.On("EntityByID", cg.CollectionID, mock.Anything).Return().Once()
+			wg.Done()
```

Comment:
this isn't called in a callback anymore, so it's called synchronously. Is the check that was done in the `EntityByID` mock not needed anymore? If so, we can just get rid of the waitgroup

Reply:
[60d0b75](https://github.com/onflow/flow-go/pull/6014/commits/60d0b75155ad2090b0d083371d76d9166d833e99)

I updated the calling of `EntityByID` and retained the waitgroup to ensure that all blocks are processed before the test ends.